### ISSUE TITLE
Feature Request: Start on bootup

### ISSUE THREAD

**Issue by berryelectronics** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#996]
Just checked the Settings, dont find a "Start with System Start" option. Would be neat i think. My Music player is generally something I use all the time when im on my pc, so a start on bootup would be really great.

**Comment by berryelectronics** [comment]
Adding the String `"C:\Users\matthias\AppData\Local\GPMDP_3\app-3.2.1\Google%20Play%20Music%20Desktop%20Player.exe"` to the Registry HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Run works fine. Only thing to look it how to script it in the installer and swap change the user name out for the currents users name.

**Comment by jostrander** [comment]
We can probably use something like this: https://github.com/ironSource/node-regedit but I'm not sure if it's something that will prompt for UAC or not.

**Comment by berryelectronics** [comment]
@jostrander elsewise Using `HKEY_CURRENT_USER` should not prompt UAC AFAIK.

**Comment by jostrander** [comment]
Agreed, because its under user scope, only HKEY_SYSTEM or w/e should cause UAC, or be rejected altogether with no prompt without elevation. You want to take a stab at implementing that? I don't run windows natively.

**Comment by berryelectronics** [comment]
I could give it a try.

**Comment by MarshallOfSound** [comment]
As a place to start looking

https://github.com/Teamwork/node-auto-launch

Just be careful on windows as we NEVER want to prompt for UAC or need UAC to do anything.  Instead of using the registry there is a folder we can use

`%APPDATA\Microsoft\Windows\Start Menu\Programs\Startup`

**Comment by berryelectronics** [comment]
Doesnt the Startup folder trap many antivirus programs because its known that many viruses use that folder?

**Comment by MarshallOfSound** [comment]
@berryelectronics I am not aware of AV targeting the startup folder and in fact Slack is one of my applications that has put itself in there.

**Comment by berryelectronics** [comment]
@MarshallOfSound Just a sidenote. Im trying to work it out, but im having a problem to install nslog. Its always exiting with the error 1. Is it just me or did you also had problems with it?

**Comment by MarshallOfSound** [comment]
@berryelectronics It's not just you, but I don't have that issue.  Basically ensure you have permissions that nslog wants and you have the required c++ packs installed.  Easiest way to get those is to install Visual Studio

**Comment by berryelectronics** [comment]
@MarshallOfSound I got Nodejs and npm with VS Community 2015, but still dont get that.

**Comment by MarshallOfSound** [comment]
What exactly is the error? Copy - Paste it

**Comment by berryelectronics** [comment]
`npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated graceful-fs@2.0.3: graceful-fs v3.0.0 and before will fail on node releases >= v6.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
npm WARN deprecated graceful-fs@3.0.8: graceful-fs v3.0.0 and before will fail on node releases >= v6.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
npm WARN deprecated lodash@1.0.2: lodash@<3.0.0 is no longer maintained. Upgrade to lodash@^4.0.0.
npm WARN deprecated graceful-fs@1.2.3: graceful-fs v3.0.0 and before will fail on node releases >= v6.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
npm WARN prefer global coffee-script@1.10.0 should be installed with -g
npm WARN prefer global npm@3.8.8 should be installed with -g`

after nslog node-gyp rebuild

`C:\Users\matthias\Documents\Git\Google-Play-Music-Desktop-Player-UNOFFICIAL\node_modules\nslog>if not defined npm_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild )
gyp ERR! configure error
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at failNoPython (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:401:14)
gyp ERR! stack     at C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:356:11
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:117:15)
gyp ERR! System Windows_NT 10.0.10586
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\matthias\Documents\Git\Google-Play-Music-Desktop-Player-UNOFFICIAL\node_modules\nslog
gyp ERR! node -v v6.0.0
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok
google-play-music-desktop-player@3.2.4 C:\Users\matthias\Documents\Git\Google-Play-Music-Desktop-Player-UNOFFICIAL`

Listing all the npm packages:
`-- UNMET PEER DEPENDENCY chai@3.5.0`
`+-- UNMET PEER DEPENDENCY eslint-plugin-jsx-a11y@^1.0.2
+-- UNMET PEER DEPENDENCY eslint-plugin-react@^5.0.1`

`npm WARN optional Skipping failed optional dependency //electron-installer-debian:
npm WARN notsup Not compatible with your operating system or architecture: electron-installer-debian@0.2.0
npm WARN optional Skipping failed optional dependency //electron-installer-redhat:
npm WARN notsup Not compatible with your operating system or architecture: electron-installer-redhat@0.2.0
npm WARN optional Skipping failed optional dependency //mouse-forward-back:
npm WARN notsup Not compatible with your operating system or architecture: mouse-forward-back@1.0.1
npm WARN optional Skipping failed optional dependency //dbus:
npm WARN notsup Not compatible with your operating system or architecture: dbus@0.2.18
npm WARN optional Skipping failed optional dependency /mpris-service/dbus:
npm WARN notsup Not compatible with your operating system or architecture: dbus@0.2.18
npm WARN eslint-config-airbnb@8.0.0 requires a peer of eslint-plugin-jsx-a11y@^1.0.2 but none was installed.
npm WARN eslint-config-airbnb@8.0.0 requires a peer of eslint-plugin-react@^5.0.1 but none was installed.
npm WARN chai-fs@0.1.0 requires a peer of chai@>= 1.6.1 < 2 but none was installed.`

`npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v6.0.0
npm ERR! npm  v3.8.6
npm ERR! code ELIFECYCLE

npm ERR! nslog@3.0.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the nslog@3.0.0 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the nslog package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs nslog
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls nslog
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\matthias\Documents\Git\Google-Play-Music-Desktop-Player-UNOFFICIAL\npm-debug.log`

So guess one part of the problem could be my missing python installation?

**Comment by MarshallOfSound** [comment]
Lol, read the error message

<pre>
Can't find Python executable "python"
</pre>

Try installing python :+1:

**Comment by berryelectronics** [comment]
Ok, got that working.
Added the feature and with a test start (npm start) it seemed like it added/removed the necessary listing in the autostart menue in the task manager. gonna try to build it and test if it really works. If it does im gonna clean my fork and just add that functionality and make a pull request. :)

**Comment by berryelectronics** [comment]
[Things are looking nice :^)](http://puu.sh/oDJ44/778f7eed24.png)

**Comment by MarshallOfSound** [comment]
@berryelectronics Sweet as, I just commented on some of your commits to target a few issues before PR :+1:

**Comment by wirespot** [comment]
For Linux distributions that conform to FreeDesktop.Org (FDO) specs, autostart can be done by creating a .desktop file in `~/.config/autostart`. Since you already add an entry to the global app menu you may already be familiar with .desktop file format. 

Nevertherless, here's an example, and [the spec is here](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html).

```
[Desktop Entry]  
Type=Application
Name=Google Play Music Desktop Player
Name[en_US]=Google Play Music Desktop Player
GenericName=google-play-music-desktop-player
Comment=Cross platform desktop player for Google Play Music.
Comment[en_US]=Cross platform desktop player for Google Play Music.
Icon=/usr/share/pixmaps/google-play-music-desktop-player.png
Exec=/usr/bin/google-play-music-desktop-player
Hidden=false
NoDisplay=false
X-GNOME-Autostart-enabled=true
```

**Comment by berryelectronics** [comment]
@wirespot [to good that auto-launch is already doing that ](https://github.com/Teamwork/node-auto-launch/blob/master/src/AutoLaunchLinux.coffee) would be nice if you could clone my PR #1017 and build it locally to check if its working. thanks :)

**Comment by MarshallOfSound** [comment]
Implemented in #1017
